### ISSUE TITLE
Fresh install errors when websocket connection is opened.

### ISSUE THREAD

**Issue by fake-name** [miguelgrinberg/python-socketio#12]
I just did a fresh install in my testing VM, and now every time I try to establish a websocket connection, it fails:

```
192.168.1.106 - - [29/Dec/2015 16:20:43] "GET /socket.io/?EIO=3&transport=websocket HTTP/1.1" 500 -
Error on request:
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/dist-packages/werkzeug/serving.py", line 194, in run_wsgi
    execute(self.server.app)
  File "/usr/local/lib/python3.4/dist-packages/werkzeug/serving.py", line 182, in execute
    application_iter = app(environ, start_response)
  File "/usr/local/lib/python3.4/dist-packages/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python3.4/dist-packages/flask_socketio/__init__.py", line 37, in __call__
    start_response)
  File "/usr/local/lib/python3.4/dist-packages/engineio/middleware.py", line 47, in __call__
    return self.engineio_app.handle_request(environ, start_response)
  File "/usr/local/lib/python3.4/dist-packages/socketio/server.py", line 303, in handle_request
    return self.eio.handle_request(environ, start_response)
  File "/usr/local/lib/python3.4/dist-packages/engineio/server.py", line 217, in handle_request
    transport, b64)
  File "/usr/local/lib/python3.4/dist-packages/engineio/server.py", line 308, in _handle_connect
    s.handle_get_request(environ, start_response)
  File "/usr/local/lib/python3.4/dist-packages/engineio/socket.py", line 75, in handle_get_request
    start_response)
  File "/usr/local/lib/python3.4/dist-packages/engineio/socket.py", line 108, in _upgrade_websocket
    self.server.async['websocket_class'])
TypeError: getattr(): attribute name must be string
```

I'm going to see if I can figure out what's going on, but this is on a fresh install via `pip3 install python-socketio`

**Comment by fake-name** [comment]
It's possibly worth noting that I'm having some other issues with virtualbox crashing a lot today, and it's not inconceivable that this is a system problem acting up in an esoteric manner.

**Comment by fake-name** [comment]
Offending call: 

```
        websocket_class = getattr(self.server.async['websocket'],
                                  self.server.async['websocket_class'])

```

Dumping the state of `self.server.async` at that point: 

```
{'thread_class': 'Thread', 'websocket': None, 'threading': <module 'threading' from '/usr/lib/python3.4/threading.py'>, 'websocket_class': None, 'queue_class': 'Queue', 'queue': <module 'queue' from '/usr/lib/python3.4/queue.py'>}
```

So the issue is that `self.server.async['websocket_class']` is `None`, and it needs to be a string at that point.

**Comment by fake-name** [comment]
Ok, never mind, I'm dumb. I forgot to install `eventlet`.

A nicer error message would be nice, but I was missing the "hey, this shit isn't installed" when the app was starting, because I do a bunch of startup and it outputs lots of logging.

**Comment by miguelgrinberg** [comment]
Are you setting `async_mode='eventlet'` in your constructor? Because normally you shouldn't, then the extension will start in the most appropriate way, depending on packages installed. If you didn't have eventlet or gevent, it would have started in threading mode, which sucks for performance, but at least works.

Will see about adding better error messages when stuff is missing, that is a fair point regardless.

**Comment by fake-name** [comment]
I'm not setting anything in my constructor (`socketio = SocketIO(app)`), but I am disallowing non-websocket connections in the javascript end.

The update timing rate I'm aiming for doesn't really work at all under long-polling, so I'd rather fail hard then have degraded performance.

**Comment by miguelgrinberg** [comment]
Ah, perfect. That's another interesting use case for which I should produce better error messages. Thanks!

**Comment by miguelgrinberg** [comment]
Fixed in python-engineio 0.8.6.
